src/pd/util.rs

```diff
@@ -177,6 +176,13 @@ impl LeaderClient {
         {
             let mut inner = self.inner.wl();
             let (tx, rx) = client.region_heartbeat().unwrap();
+            warn!("heartbeat sender and receiver are stale, refreshing..");
+
+            // Try to cancel an unused heartbeat sender.
+            if let Either::Left(Some(ref mut r)) = inner.hb_sender {
```

Comment:
So, the bug is that if we reconnect twice before the previous sender is used, then the previous sender will not be canceled, right?

Reply:
Yes, you are right.